test/services/EnrolmentStoreCachingServiceSpec.scala

```diff
@@ -50,43 +53,39 @@ class EnrolmentStoreCachingServiceSpec extends BaseSpec with MockitoSugar with S
         mockSessionCache.fetchAndGetEntry[SelfAssessmentUserType](any())(any(), any(), any())
       ) thenReturn Future.successful(fetchResult)
 
+      when(mockEnrolmentsConnector.getUserIdsWithEnrolments(any())(any(), any()))
+        .thenReturn(Future.successful(connectorResult))
+
       c
     }
   }
 
-  val userTypeList: List[(SelfAssessmentUserType, String)] = List(
-    (ActivatedOnlineFilerSelfAssessmentUser(SaUtr("111111111")), "an Activated SA user"),
-    (NotYetActivatedOnlineFilerSelfAssessmentUser(SaUtr("111111111")), "a Not Yet Activated SA user"),
-    (WrongCredentialsSelfAssessmentUser(SaUtr("111111111")), "a Wrong credentials SA user"),
-    (NotEnrolledSelfAssessmentUser(SaUtr("111111111")), "a Not Enrolled SA user"),
-    (NonFilerSelfAssessmentUser, "a Non Filer SA user")
-  )
+  "EnrolmentStoreCachingHelper" when {
 
-  "EnrolmentStoreCachingHelper" should {
+    "when the cache is empty and the connector is called" should {
 
-    userTypeList.foreach {
-      case (userType, key) =>
-        s"add $key to the cache" in new LocalSetup {
+      val saUtr = SaUtr("111111111")
 
-          override val cacheResult: CacheMap =
-            CacheMap("id", Map(SelfAssessmentUserType.cacheId -> Json.toJson(userType)))
+      "return NonFilerSelfAssessmentUser when the connector returns a Left" in new LocalSetup {
 
-          sut.addSaUserTypeToCache(userType).futureValue shouldBe cacheResult
-        }
+        override val connectorResult: Either[String, Seq[String]] = Left("An error has occurred")
 
-        s"fetch $key from the cache" in new LocalSetup {
+        sut.getSaUserTypeFromCache(saUtr).futureValue shouldBe NonFilerSelfAssessmentUser
+      }
 
-          override val fetchResult: Option[SelfAssessmentUserType] = Some(userType)
+      "return NotEnrolledSelfAssessmentUser when the connector returns a Right with an empty sequence" in new LocalSetup {
 
-          sut.getSaUserTypeFromCache().futureValue shouldBe fetchResult
-        }
-    }
+        override val connectorResult: Either[String, Seq[String]] = Right(Seq[String]())
 
-    "return None when the cache is empty" in new LocalSetup {
+        sut.getSaUserTypeFromCache(saUtr).futureValue shouldBe NotEnrolledSelfAssessmentUser(saUtr)
+      }
 
-      sut.getSaUserTypeFromCache().futureValue shouldBe None
-    }
+      "return WrongCredentialsSelfAssessmentUser when the connector returns a Right with a non-empty sequence" in new LocalSetup {
 
-  }
+        override val connectorResult: Either[String, Seq[String]] = Right(Seq[String]("Hello there"))
 
+        sut.getSaUserTypeFromCache(saUtr).futureValue shouldBe WrongCredentialsSelfAssessmentUser(saUtr)
+      }
+    }
+  }
 }
```

Comment:
Missing a test on the cache, basically `verify` call on` enrolmentsConnector.getUserIdsWithEnrolments` to determine when something has actually been cached.

So the test would be named something like "calling same function twice, only calls the cache once"